tests/test_eeprom.c

```diff
@@ -0,0 +1,86 @@
+    memcpy( pif_recv_buf, &pif_out[1], sizeof(pif_recv_buf) );
+    ASSERT_EQUAL_HEX(pif_out[0], pif_in[0], "pif response mismatch");
+    if (pif_recv_buf[1] == 0xc0) {
+        LOG( "16K EEPROM detected.\n" );
+        ASSERT_EQUAL_HEX(pif_out[1], (uint64_t)0x00c000fe00000000, "pif response mismatch");
+        ASSERT(eeprom_type == EEPROM_16K, "eeprom type changed?");
+    } else if (pif_recv_buf[1] == 0x80) {
+        LOG( "4K EEPROM detected.\n" );
+        ASSERT_EQUAL_HEX(pif_out[1], (uint64_t)0x008000fe00000000, "pif response mismatch");
+        ASSERT(eeprom_type == EEPROM_4K, "eeprom type changed?");
+    } else {
+        LOG( "EEPROM not detected.\n" );
+        ASSERT_EQUAL_HEX(pif_out[1], (uint64_t)0x000000fe00000000, "pif response mismatch");
+        ASSERT(eeprom_type == EEPROM_NONE, "eeprom type changed?");
+    }
+
+    // Test Joybus Info command with 0xFFFFFF recv buf
+    pif_in[0] = 0x00000000ff0103ff;
+    pif_in[1] = 0xfffffffe00000000;
+    pif_execute( pif_in, sizeof(pif_in), pif_out, sizeof(pif_out) );
+    memcpy( pif_recv_buf, &pif_out[1], sizeof(pif_recv_buf) );
+    ASSERT_EQUAL_HEX(pif_out[0], pif_in[0], "pif response mismatch");
+    if (pif_recv_buf[1] == 0xc0) {
+        LOG( "16K EEPROM detected.\n" );
+        ASSERT_EQUAL_HEX(pif_out[1], (uint64_t)0x00c000fe00000000, "pif response mismatch");
+        ASSERT(eeprom_type == EEPROM_16K, "eeprom type changed?");
+    } else if (pif_recv_buf[1] == 0x80) {
+        LOG( "4K EEPROM detected.\n" );
+        ASSERT_EQUAL_HEX(pif_out[1], (uint64_t)0x008000fe00000000, "pif response mismatch");
+        ASSERT(eeprom_type == EEPROM_4K, "eeprom type changed?");
+    } else {
+        LOG( "EEPROM not detected.\n" );
+        ASSERT_EQUAL_HEX(pif_out[1], (uint64_t)0xfffffffe00000000, "pif response mismatch");
+        ASSERT(eeprom_type == EEPROM_NONE, "eeprom type changed?");
+    }
+
+    // Test Joybus Info command with 0x000000 recv buf
+    pif_in[0] = 0x00000000ff0103ff;
+    pif_in[1] = 0x000000fe00000000;
+    pif_execute( pif_in, sizeof(pif_in), pif_out, sizeof(pif_out) );
+    memcpy( pif_recv_buf, &pif_out[1], sizeof(pif_recv_buf) );
+    ASSERT_EQUAL_HEX(pif_out[0], pif_in[0], "pif response mismatch");
+    if (pif_recv_buf[1] == 0xc0) {
+        LOG( "16K EEPROM detected.\n" );
+        ASSERT_EQUAL_HEX(pif_out[1], (uint64_t)0x00c000fe00000000, "pif response mismatch");
+        ASSERT(eeprom_type == EEPROM_16K, "eeprom type changed?");
+    } else if (pif_recv_buf[1] == 0x80) {
+        LOG( "4K EEPROM detected.\n" );
+        ASSERT_EQUAL_HEX(pif_out[1], (uint64_t)0x008000fe00000000, "pif response mismatch");
+        ASSERT(eeprom_type == EEPROM_4K, "eeprom type changed?");
+    } else {
+        LOG( "EEPROM not detected.\n" );
+        ASSERT_EQUAL_HEX(pif_out[1], (uint64_t)0x000000fe00000000, "pif response mismatch");
+        ASSERT(eeprom_type == EEPROM_NONE, "eeprom type changed?");
+    }
+}
```

Comment:
I'm not sure how I feel about this test. This sounds more like a test to help emulator authors. Libdragon tests should be tests of libdragon functionality to make sure it keeps working over time. For instance, I would expect here a test of `eeprom_read_bytes` / `eeprom_write_bytes`. This wouldn't required to export `pif_execute` BTW.

Reply:
See https://github.com/DragonMinded/libdragon/pull/125#discussion_r658879871

Reply:
`eeprom_read_bytes` and `eeprom_write_bytes` get exercised by the EEPROM filesystem tests, so I didn't see a need to include them here